src/cml.js

```diff
@@ -0,0 +1,103 @@
+const { execSync } = require('child_process');
+const git_url_parse = require('git-url-parse');
+
+const Gitlab = require('./drivers/gitlab');
+const Github = require('./drivers/github');
+const { upload, exec } = require('./utils');
+
+const uri_no_trailing_slash = (uri) => {
+  return uri.endsWith('/') ? uri.substr(0, uri.length - 1) : uri;
+};
+
+const repo_from_origin = () => {
+  const origin = execSync('git config --get remote.origin.url').toString(
+    'utf8'
+  );
+  return git_url_parse(origin).toString('https').replace('.git', '');
+};
+
+const infer_driver = (opts = {}) => {
+  const { repo } = opts;
+  if (repo && repo.includes('github.com')) return 'github';
+  if (repo && repo.includes('gitlab.com')) return 'gitlab';
```

Comment:
Could these checks be just a bit more rigorous?
For example, I might have a local repo in my file system like `~/src/my-gitlab.com-repo/` which might be no GitLab repo.

Reply:
repo has to be an URL. Its the url to be used by the API. can not be anything else and when its going to be used it will fail if not.